jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveSqlGenerationVisitor.java

```diff
@@ -33,86 +31,94 @@ public HiveSqlGenerationVisitor(final SqlDialect dialect, final SqlGenerationCon
-                if(node.getType().equals(SqlNodeType.COLUMN)) {
-                    final SqlColumn sqlColumn = (SqlColumn) node;
-                    final String typeName = ColumnAdapterNotes.deserialize(sqlColumn.getMetadata().getAdapterNotes(), sqlColumn.getMetadata().getName()).getTypeName();
-                    if (typeName.equals("BINARY")) {
-                        selectListElements.add("base64(" + node.accept(this) + ")");
-                    } else {
-                        selectListElements.add(node.accept(this));
-                    }
+        }
+        return String.join(", ", selectListElements);
+    }
+
+    private List<String> getSelectStarList(final SqlSelectList selectList) throws AdapterException {
+        final List<String> selectListElements = new ArrayList<>();
+        if (SqlGenerationHelper.selectListRequiresCasts(selectList, nodeRequiresCast)) {
+            final List<TableMetadata> tableMetadata = new ArrayList<>();
+            final SqlStatementSelect select = (SqlStatementSelect) selectList.getParent();
+            SqlGenerationHelper.addMetadata(select.getFromClause(), tableMetadata);
+            addColumns(selectListElements, tableMetadata);
+        } else {
+            selectListElements.add("*");
+        }
+        return selectListElements;
+    }
+
+    private void addColumns(final List<String> selectListElements, final List<TableMetadata> tableMetadata)
+            throws AdapterException {
+        int columnId = 0;
+        for (final TableMetadata tableMeta : tableMetadata) {
+            for (final ColumnMetadata columnMeta : tableMeta.getColumns()) {
+                final SqlColumn sqlColumn = new SqlColumn(columnId, columnMeta);
+                final String typeName = ColumnAdapterNotes
+                        .deserialize(sqlColumn.getMetadata().getAdapterNotes(), sqlColumn.getMetadata().getName())
+                        .getTypeName();
+                if (typeName.equals(BINARY_TYPE_NAME)) {
+                    selectListElements.add("base64(" + super.visit(sqlColumn) + ")");
+                } else {
+                    selectListElements.add(super.visit(sqlColumn));
                 }
-                else{
+                ++columnId;
+            }
+        }
+    }
+
+    private List<String> getSelectList(final SqlSelectList selectList) throws AdapterException {
+        final List<String> selectListElements = new ArrayList<>();
+        for (final SqlNode node : selectList.getExpressions()) {
+            if (node.getType().equals(SqlNodeType.COLUMN)) {
+                final SqlColumn sqlColumn = (SqlColumn) node;
+                final String typeName = ColumnAdapterNotes
+                        .deserialize(sqlColumn.getMetadata().getAdapterNotes(), sqlColumn.getMetadata().getName())
+                        .getTypeName();
+                if (typeName.equals(BINARY_TYPE_NAME)) {
+                    selectListElements.add("base64(" + node.accept(this) + ")");
```

Comment:
We should really make this the standard behavior for all binary types in the base implementation. Add an issue please.

Reply:
done